component-library/component-lib/src/lib/icon-button/icon-button.component.ts

```diff
@@ -1,18 +1,73 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, Input, Output, EventEmitter, OnInit} from '@angular/core';
 export enum IconButtonCategories {
   primary = 'primary',
   critical = 'critical',
   custom = 'custom'
 }
+
+export enum IconButtonSize {
+  large = 'large',
+  small = 'small',
+  extraSmall = 'extraSmall'
+}
+
+export interface IconButtonIconConfig {
+  class: string; // Fontawesome icon class
+  color?: string; // icon color
+}
+
+export interface IconButtonComponentConfig {
```

Comment:
All interfaces must start with 'I', meaning this should be IIconButtonComponentConfig